### ISSUE TITLE
append method of MapAxis and LabelMapAxis is a hidden classmethod. 

### ISSUE THREAD

**Issue by MRegeard** [gammapy/gammapy#4443]
The append method of `MapAxis` and `LabelMapAxis` returns a new object with the appened axis. 

I think that this is confusing since the append terminology is often used to describe a methods that add an element at the end of an object. 

Should it stay like that? Should it be a private method ?

**Comment by adonath** [comment]
Thanks @MRegeard, I see what you mean. I think you are referring to the pattern that lists in python are appended "in place".

I think appending `MapAxis` in place is not a good option, because the pattern to return the modified object is consistent within Gammapy, with very well justified exceptions such as `.stack()` which acts in place. Also note that e.g.https://numpy.org/doc/stable/reference/generated/numpy.append.html does not operate in place either.

If we agree it is an improvement, we could consider deprecating `.append()` and e.g. rename to `.concatenate()` or similar.

**Comment by adonath** [comment]
Actually I think I actually agree on the rename, because we have e.g. `Models.append()` and `Datasets.append()` which act in place.

**Comment by MRegeard** [comment]
I also agree with the rename. I think it would be less confusing, and as you pointed out, more consistent with respect to other append in gammapy.

**Comment by MRegeard** [comment]
Since this is a rename, we should probably have a deprecation period right ?

**Comment by registerrier** [comment]
Yes, but the decorator allows you change it directly this way:
```
from gammapy.utils.deprecation import deprecated_attribute

class some_class:
    method = deprecated_attribute("method", "1.1", alternative="function")
    
    def function(self, argument):
        print(f"argument is {argument}")

some_class().method("whatever")
```

**Comment by registerrier** [comment]
This is implemented in #4506 . Closing.
